test/kbaseauthcontroller/controller.go

```diff
@@ -119,50 +125,83 @@ func waitForStartup(port string) error {
 	return startupErr
 }
 
-func getClassPath(jarsDir string) (string, error) {
-	jarsDir, err := filepath.Abs(jarsDir)
+func getClassPath(auth2Jar string) (string, error) {
+	jpath, err := filepath.Abs(auth2Jar)
 	if err != nil {
 		return "", err
 	}
-	cp := []string(nil)
-	for _, j := range jars { // global variable, yech
-		jpath := path.Join(jarsDir, j)
-		if _, err := os.Stat(jpath); os.IsNotExist(err) {
-			return "", fmt.Errorf("Jar %v does not exist", jpath)
-		}
-		cp = append(cp, jpath)
+	if _, err := os.Stat(jpath); os.IsNotExist(err) {
+		return "", fmt.Errorf("jar %v does not exist", jpath)
 	}
-	return strings.Join(cp, ":"), nil
+	return jpath, nil
 }
 
-func installTemplates(jarsDir string, templateDir string) error {
-	templateZip := path.Join(jarsDir, authTemplates)
-	arch, err := zip.OpenReader(templateZip) // global variable, yech
+func pullTemplatesOutofAuth2Jar(classPath string) (string, error) {
+	dirPath := filepath.Dir(classPath)
+	outfile, err := os.Create(filepath.Join(dirPath, "output.txt"))
 	if err != nil {
-		return err
+		return "", err
+	}
+
+	cmdargs := []string{classPath, "-d", dirPath}
+	cmd := exec.Command("unzip", cmdargs...)
+	cmd.Stdout = outfile
+	cmd.Stderr = outfile
+	err = cmd.Start()
+	if err != nil {
+		return "", err
 	}
-	for _, f := range arch.File {
-		name := f.FileHeader.Name
+
+	tpath := filepath.Join(dirPath, "kbase_auth2_templates")
+
+	i := 0
+	for i < attempts {
+		if _, err := os.Stat(tpath); os.IsNotExist(err) {
+			time.Sleep(1 * time.Second)
+			i += 1
+			continue
+		}
+		break
+    }
+	if i == attempts {
+		return "", fmt.Errorf("the template folder %v does not exist", tpath)
+	}
+	return tpath, err
+}
+
+func installTemplates(tpath string, templateDir string) error {
+	files, err := ioutil.ReadDir(tpath)
+	if err != nil {
+        return err
+    }
+	for _, f := range files {
+		name := f.Name()
 		if !strings.HasSuffix(name, "/") { // not a directory
 			name = path.Clean(name)
 			if path.IsAbs(name) || strings.HasPrefix(name, "..") {
```

Comment:
This should be impossible when reading a directory right? This step, if done, needs to be done while reading the zip file so that a malicious entry doesn't overwrite something outside the target directory

Reply:
👍

Reply:
:+1: